src/transport/SecureSessionMgr.h

```diff
@@ -181,8 +181,8 @@ class DLL_EXPORT SecureSessionMgr : public TransportMgrDelegate
      *   to retransmit it, will be stored in *bufferRetainSlot.
      */
     CHIP_ERROR SendMessage(SecureSessionHandle session, System::PacketBufferHandle && msgBuf);
-    CHIP_ERROR SendMessage(SecureSessionHandle session, PayloadHeader & payloadHeader, System::PacketBufferHandle && msgBuf,
-                           EncryptedPacketBufferHandle * bufferRetainSlot = nullptr);
+    CHIP_ERROR SendMessage(SecureSessionHandle session, PayloadHeader & payloadHeader, PacketHeader & packetHeader,
```

Comment:
Could you add a comment here about what each argument is for?

I am a bit worried that we start adding too many parameters here. 5 is already a lot.

Also worried about separation of concerns: I imagine we want objects to have a clear control. Should we instead set some sending flags? Passing in an entire packet header that is generally fully managed by the secure session (secure session defines lengths, encryptions and such) seems too much. We are sending an entire object to allow the caller to set a single bit flag.

Reply:
Added, we already have the sending flags defined in the messaging layer to simply the configuration. 
https://github.com/project-chip/connectedhomeip/blob/master/src/messaging/Flags.h#L60

The clients of messaging layer only need to configure the SendFlags, PayloadHeader and PacketHeader are defined in transport layer which are aligned with spec. Currently, PacketHeader is always hard coded to default value, which is not exposed to upper layer to configure.